reccmp/tools/aggregate.py

```diff
@@ -127,6 +129,15 @@ def main():
             "expected arguments for --output, --html, or --diff. (No output action specified)"
         )
 
+    if args.diff and len(args.diff) == 1 and not args.samples:
+        parser.error("expected two report files to diff")
+
+    return args
+
+
+def main():
+    args = parse_args(sys.argv[1:])
```

Comment:
I'd still pass all arguments, and split inside `parse_args`.
The [`prog` argument](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser) can then get the program name, which is used for `--help`.